### PR TITLE
Use Eigen in enkf_linalg

### LINKED ISSUE #3040: Use Eigen in enkf_linalg

### PR DESCRIPTION
**Pull request by dafeda**
**Issue**
Resolves #3040

## Pre review checklist

- [ x] Added appropriate labels

Adding labels helps the maintainers when writing release notes, see sections and the
corresponding labels here: https://github.com/equinor/ert/blob/main/.github/release.yml

### CODE REVIEW

File: libres/lib/analysis/enkf_linalg.cpp [hunk 1]
```diff
@@ -30,12 +30,11 @@ Eigen::MatrixXd enkf_linalg_genX3(const Eigen::MatrixXd &W,
     return X3;
 }
 
-static int enkf_linalg_num_significant(int num_singular_values,
-                                       const double *sig0, double truncation) {
+static int enkf_linalg_num_significant(const Eigen::VectorXd singular_values,
```
Comment (pinkwah) [inline libres/lib/analysis/enkf_linalg.cpp:33]:
```suggestion
static int enkf_linalg_num_significant(const Eigen::VectorXd &singular_values,
```

File: libres/lib/analysis/enkf_linalg.cpp [hunk 2]
```diff
@@ -57,39 +56,28 @@ static int enkf_linalg_num_significant(int num_singular_values,
...
-        for (i = 0; i < num_significant; i++)
-            inv_sig0[i] = 1.0 / sig0[i];
-
-        /* Explicitly setting the insignificant singular values to zero. */
-        for (i = num_significant; i < num_singular_values; i++)
-            inv_sig0[i] = 0;
+            singular_values, std::get<double>(truncation));
     }
 
+    /* Inverting the significant singular values */
```
Comment (pinkwah) [inline libres/lib/analysis/enkf_linalg.cpp:76]: Unnecessary comment

File: libres/tests/analysis/test_update.cpp [hunk 3]
```diff
@@ -313,10 +313,10 @@ SCENARIO("Running analysis update with and without row scaling on linear model",
...
                 Eigen::MatrixXd A_no_scaling_T = A_no_scaling.transpose();
                 Eigen::MatrixXd A_prior_T = A.transpose();
-                REQUIRE(matrix_columns_equal(&A_with_scaling_T, 0,
-                                             &A_no_scaling_T, 0));
-                REQUIRE(
-                    matrix_columns_equal(&A_with_scaling_T, 1, &A_prior_T, 1));
+                REQUIRE(A_with_scaling_T(Eigen::all, 0)
+                            .isApprox(A_no_scaling_T(Eigen::all, 0)));
+                REQUIRE(A_with_scaling_T(Eigen::all, 1)
+                            .isApprox(A_prior_T(Eigen::all, 1)));
```
Comment (pinkwah) [inline libres/tests/analysis/test_update.cpp:319]: Consider
```suggestion
                REQUIRE(A_with_scaling_T.col(0)
                            .isApprox(A_no_scaling_T.col(0)));
                REQUIRE(A_with_scaling_T.col(1)
                            .isApprox(A_prior_T.col(1)));
```

Comment (dafeda) [reply libres/tests/analysis/test_update.cpp]: Pretty 👍

**Review by pinkwah — APPROVED** [review]

**Pull request merged**